CMakeLists.txt

```diff
@@ -174,6 +181,72 @@ else()
   set(FIRESTORE_SOURCE_DIR ${FIREBASE_BINARY_DIR}/external/src/firestore)
 endif()
 
+if(DESKTOP)
+  # Use the static versions of the OpenSSL libraries.
+  set(OPENSSL_USE_STATIC_LIBS TRUE)
+  if (MSVC)
+    # Get the correct version of the OpenSSL libraries based on building for MT.
+    if ("${CMAKE_CXX_FLAGS_DEBUG}" MATCHES "/MT" OR
+        "${CMAKE_CXX_FLAGS_RELEASE}" MATCHES "/MT")
+      set(OPENSSL_MSVC_STATIC_RT TRUE)
+    else()
+      set(OPENSSL_MSVC_STATIC_RT FALSE)
+    endif()
+  endif()
+
+  # Use BoringSSL instead of OpenSSL.
+  set(BORINGSSL_ROOT_DIR ${CMAKE_BINARY_DIR}/external/src/boringssl/src CACHE STRING "" FORCE)
+  set(BORINGSSL_BINARY_DIR ${CMAKE_BINARY_DIR}/external/src/boringssl-build CACHE STRING "" FORCE)
+  set(OPENSSL_ROOT_DIR ${BORINGSSL_ROOT_DIR} CACHE STRING "" FORCE)
+
+  # The call below to build_external_dependencies will make sure that these
+  # libraries exist before the libraries are imported via add_library.
+  if (MSVC)
+    set(OPENSSL_SSL_LIBRARY ${BORINGSSL_BINARY_DIR}/ssl/${CMAKE_BUILD_TYPE}/ssl.lib CACHE FILEPATH "" FORCE)
+    set(OPENSSL_CRYPTO_LIBRARY ${BORINGSSL_BINARY_DIR}/crypto/${CMAKE_BUILD_TYPE}/crypto.lib CACHE FILEPATH "" FORCE)
+  else()
+    set(OPENSSL_SSL_LIBRARY ${BORINGSSL_BINARY_DIR}/ssl/libssl.a CACHE FILEPATH "" FORCE)
+    set(OPENSSL_CRYPTO_LIBRARY ${BORINGSSL_BINARY_DIR}/crypto/libcrypto.a CACHE FILEPATH "" FORCE)
+  endif()
+endif()
+
+if(NOT FIREBASE_CPP_USE_PRIOR_GRADLE_BUILD)
+  message(STATUS "Building external project dependencies...")
```

Comment:
nit: Something that wasn't immediately clear was why we have this condition for building external dependencies. Also, isn't boringssl required only for desktop builds? Maybe a one line comment explaining the case might help.

Reply:
This was leftover from when this call was right after the download, but I moved the external build into the if(DESKTOP) instead, since we only need it on desktop anyway.